week-1/Homework/mandatory/5-magic-8-ball.js

```diff
@@ -45,22 +45,60 @@ Very doubtful.
 
 // This should log "The ball has shaken!"
 // and return the answer.
-function shakeBall() {}
+function shakeBall() {
+  var ball = console.log("The ball has shaken!");
+  return "The ball has shaken!";
+}
 
 // The answer should come from shaking the ball
-let answer;
+let answer = [
+  "It is certain",
+  "It is decidedly so",
+  "Without a doubt",
+  "Yes - definitely",
+  "You may rely on it",
+  "As I see it, yes",
+  "Most likely",
+  "Outlook good",
+  "Yes",
+  "Signs point to yes",
+  "Reply hazy, try again",
+  "Ask again later",
+  "Better not tell you now",
+  "Cannot predict now",
+  "Concentrate and ask again",
+  "Don't count on it",
+  "My reply is no",
+  "My sources say no",
+  "Outlook not so good",
+  "Very doubtful",
+];
 
 // When checking the answer, we should tell someone if the answer is
 // - very positive
 // - positive
 // - negative
 // - very negative
-function checkAnswer() {}
+
+function checkAnswer() {
+  var randomNumber = Math.random();
+  var randomAnswer = Math.floor(randomNumber * answer.length);
+
+  if (randomAnswer >= 0 && randomAnswer <= 4) {
+    return "very positive";
+  } else if (randomAnswer > 5 && randomAnswer <= 9) {
+    return "positive";
+  } else if (randomAnswer > 10 && randomAnswer <= 14) {
+    return "negative";
+  } else {
+    return "very negative";
+  }
+}
 
```

Comment:
There is some issues with this fix. you should use the result of the first function in the second one, you're really close. we will release the fix soon.